docs/tests_in_tilt.md

```diff
@@ -0,0 +1,109 @@
+
+Your tests will appear in their own section of the overview grid, and alerts will be reflected on the test cards and in the status bar, so that you know at a glance when something is wrong.
+
+<figure>
+    <img src="/assets/img/tests-in-tilt/overview-with-tests.png" alt="the Tilt UI overview page, with several tests">
+    <figcaption>Tests appear in their own section of the overview, and test status is summarized in the status bar at the top of the page</figcaption>
+</figure>
+
+Just like any other resource managed by Tilt, you can drill into a test and see logs and other details.
+
+In the log view, you can filter the sidebar by item type (if, say, you want to see only tests, or you want to filter tests out and focus on your app in the cluster). You can also toggle "Alerts On Top" sorting, to make it even more visible when something has gone wrong and needs your attention
+
+<figure>
+    <img src="/assets/img/tests-in-tilt/sidebar-alerts-on-top.png" alt="sidebar displaying only tests, with alerts sorted to the top">
+    <figcaption>Here's an example of sidebar filtering (displaying only tests). In "Alerts on Top" mode, any cards with alerts are sorted to the top of the sidebar for easy visibility</figcaption>
+</figure>
+
+### When Tests Run
+Just as with a `local_resource`, you can associate files with a test via the `deps=[...]` parameter. When any of these file dependencies change, Tilt will re-execute the test:
+
+```python
+# A test depending on an entire directory
+test('cart-tests', 'go test ./pkg/cart', deps=['./pkg/cart'])
+
+# A test depending on specific relevant files
+test('unit-test-remittance', 'pytest remittance_test.py',
+     deps=['app/remittance.py', 'app/remittance_test.py', 'app/payment_utils.py'])
+```
+(More on how to intelligently set your dependencies below.)
+
+Sometimes, you don't want your test(s) to execute automatically. Maybe you want to turn off automatic-ness for all but a few tests you're iterating on, or prevent an expensive integration test from running all the time. Use the "auto" trigger in the UI to toggle automatic execution on or off for a given test.
+
+<figure>
+    <img src="/assets/img/tests-in-tilt/auto-toggles" alt="a close-up of several cards in the sidebar, with 'auto' variously toggled on or off">
+    <figcaption>In this example, tests "foo" and "beep" are in auto mode--they will execute automatically when any of their `deps` are changed.</figcaption>
+</figure>
+
+You can also set default AUTO values for your tests via the Tiltfile, using [the `trigger_mode` parameter](https://docs.tilt.dev/manual_update_control.html).
+
+### Programmatically Registering Tests
+Part of what's great about Starlark (the dialect of Python that Tiltfiles are written in) is that it's a _programming language_, so you can use it for complex stuff. In particular, you can use it to programmatically register your tests to Tilt in whichever way works best for you. Here are some examples:
```

Comment:
```suggestion
Part of what's great about Starlark (the dialect of Python that Tiltfiles are written in) is that it's a _programming language_, which makes Tiltfiles extremely flexible. Let's look at a few of the many ways you can register your tests to Tilt, according to your tastes:
```

I don't have strong feelings on this but some vague senses of things that I tried to correct:
1. maybe just me but not a huge fan of "stuff" in this context
2. maybe better to not give the impression that what's coming up is "complex"
3. I think this usage of "in particular" is wrong, since there are lots of great complex things you can do in starlark that are not registering tests. I get the sense of what you're going for and maybe something like "in particular as far as tests are concerned, you can..." could work, but feels awkward. But also I don't know if anyone will really notice or care and I don't even know if I'd comment on this if anyone but you had written it.

Reply:
i love this comment thank you for this comment i'm glad you left it